### PR TITLE
[20.09] Include dataset and state in valid state exception

### PR DESCRIPTION
**Pull request by mvdbeek**

### CODE REVIEW

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -2773,7 +2773,7 @@ def check_dataset_instance(input_dataset):
 
             if self.require_dataset_ok:
                 if input_dataset.state != input_dataset.dataset.states.OK:
-                    raise ValueError("Tool requires inputs to be in valid state.")
+                    raise ValueError("Tool requires inputs to be in valid state, but {} is in state '{}".format(input_dataset, input_dataset.state))
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py]: Should this be
```suggestion
                    raise ValueError("Tool requires inputs to be in valid state, but dataset {} is in state '{}'".format(input_dataset, input_dataset.state))
```
?

Comment (mvdbeek) [reply lib/galaxy/tools/__init__.py]: I was hoping to get the `__repr__` of the ~HDA~ dataset

Comment (mvdbeek) [reply lib/galaxy/tools/__init__.py]: (because this may not be an HDA)

Comment (nsoranzo) [reply lib/galaxy/tools/__init__.py]: OK, I've updated the suggestion.

**Review by nsoranzo — APPROVED** [review]

**Review by dannon — APPROVED** [review]

**Pull request merged**